script-library/docker-in-docker-debian.sh

```diff
@@ -326,7 +327,8 @@ tee /usr/local/share/docker-init.sh > /dev/null \
 
 set -e
 
-AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
+export AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
+export DOCKER_DEFAULT_ADDRESS_POOL=$DOCKER_DEFAULT_ADDRESS_POOL
```

Comment:
These do not need to be exported, as they are only being used within the same generated script as they are being interpolated into here.

```suggestion
AZURE_DNS_AUTO_DETECTION=$AZURE_DNS_AUTO_DETECTION
DOCKER_DEFAULT_ADDRESS_POOL=$DOCKER_DEFAULT_ADDRESS_POOL
```

Reply:
If we don't export these, when `sudo -E /bin/sh -c "${dockerd_start}"` is run these variables are not available.

Reply:
Ok, I found a workaround here by injecting these into the `dockerd_start` variable in the generated script instead of the script directly. This allows them to get redefined when you run `dockerd_start` contents with sudo. 👍 

I think this should resolve your comment about using `sudo -E` and `export`.